### PR TITLE
[BUILDER] Implement Builder API consumer

### LINKED ISSUE #2415: [BUILDER] - Builder API consumer
### What is this task and why do we need to work on it?

For builder integration we need a tide-disco application that can consume builder API.

### What work will need to be done to complete this task?

_No response_

### Are there any other details to include?

_No response_

### What are the acceptance criteria to close this issue?

There's a builder API consumer conforming to builder API (TBD). Corresponding tests are added.

### Branch work will be merged to (if not the default branch)

_No response_

### PR DESCRIPTION
**Pull request by QuentinI**
Closes #2415

### This PR: 
- Implements a `BuilderClient` able to communicate with builders through `hs-builder-api`
- Adds a dummy builder, which "builds" empty blocks and adds a test to run `BuilderClient` against it

### This PR does not: 
- Implement more sophisticated HotShot-internal builders (which we do want to do later)
- Integrate the builder client into HotShot

### Key places to review: 
`crates/task-impls/src/builder.rs`
`crates/testing/src/block_builder.rs`
`crates/testing/tests/block_builder.rs`

### CODE REVIEW

**Comment by move47** [comment]
I am not sure whether it is right time to include the recent changes introduced in api responses structure here: https://github.com/EspressoSystems/hs-builder-api/pull/14. @nyospe 

If you want to do some development based on these recent changes, please use branch https://github.com/EspressoSystems/hs-builder-api/tree/builder_specific_information for it.

FYI, I have included these changes in the builder side.

**Comment by move47** [comment]
> I am not sure whether it is right time to include the recent changes introduced in api responses structure here: [EspressoSystems/hs-builder-api#14](https://github.com/EspressoSystems/hs-builder-api/pull/14). @nyospe
> 
> If you want to do some development based on these recent changes, please use branch https://github.com/EspressoSystems/hs-builder-api/tree/builder_specific_information for it.
> 
> FYI, I have included these changes in the builder side.

The changes has been merged in `main`. So, no need to point the a branch now.

**Review by bfish713 — APPROVED** [review]

File: crates/testing/src/block_builder.rs [hunk 1]
```diff
@@ -0,0 +1,104 @@
...
+                )
+                .unwrap(),
+                sender: self.pub_key,
+                _phantom: std::marker::PhantomData,
+            })
+        } else {
+            Err(BuildError::Missing)
+        }
+    }
+    async fn submit_txn(
```
Comment (nyospe) [inline crates/testing/src/block_builder.rs:79]: Looks like this needs to be updated to support the current version, which adds a `claim_block_header_input`.

`submit_txn` has also been moved to a different trait, AcceptsTxnSubmits<I>, to enable decoupling of locked access.

**Comment by QuentinI** [comment]
@bfish713 should be ready for merge now

**Review by bfish713 — APPROVED** [review]

**Pull request merged**